### DISCUSSION TITLE
Update AUTOMATIC1111 to 1.6.0

### DISCUSSION THREAD

**Discussion by Krisseck** [AbdBarho/stable-diffusion-webui-docker#583] [General]
AUTOMATIC1111 released 1.6.0, would be great to have it in a docker container! 

https://github.com/AUTOMATIC1111/stable-diffusion-webui/releases/tag/v1.6.0

It was RC for a while, but this is now an official release.

**Comment by cloudaxes** [comment]
Pull request here: https://github.com/AbdBarho/stable-diffusion-webui-docker/pull/585

**Comment by 66li** [comment]
The versions of these dependent libraries should also be updated
```
RUN . /clone.sh taming-transformers https://github.com/CompVis/taming-transformers.git 24268930bf1dce879235a7fddd0b2355b84d7ea6 \
  && rm -rf data assets **/*.ipynb
RUN . /clone.sh stable-diffusion-stability-ai https://github.com/Stability-AI/stablediffusion.git 47b6b607fdd31875c9279cd2f4f16b92e4ea958e \
  && rm -rf assets data/**/*.png data/**/*.jpg data/**/*.gif
RUN . /clone.sh CodeFormer https://github.com/sczhou/CodeFormer.git c5b4593074ba6214284d6acd5f1719b6c5d739af \
  && rm -rf assets inputs
RUN . /clone.sh BLIP https://github.com/salesforce/BLIP.git 48211a1594f1321b00f14c9f7a5b4813144b2fb9
RUN . /clone.sh k-diffusion https://github.com/crowsonkb/k-diffusion.git c9fe758757e022f05ca5a53fa8fac28889e4f1cf
RUN . /clone.sh clip-interrogator https://github.com/pharmapsychotic/clip-interrogator 2486589f24165c8e3b303f84e9dbbea318df83e8
RUN . /clone.sh generative-models https://github.com/Stability-AI/generative-models 5c10deee76adad0032b412294130090932317a87
```

**Comment by alex-red** [comment]
Anyone else not seeing the new samplers after upgrading to 1.6? 3M SDE Karras specifically.

**Comment by cloudaxes** [comment]
You'll need to upgrade the libraries that @66li is mentioning, specifically k-diffusion to get the 3M SDE Karras.

**Reply by alex-red** [reply]
Ah thanks, that worked.

I also went ahead and pointed all of the repos to the latest commit:
```Dockerfile
RUN . /clone.sh taming-transformers https://github.com/CompVis/taming-transformers.git 3ba01b241669f5ade541ce990f7650a3b8f65318 \
  && rm -rf data assets **/*.ipynb

RUN . /clone.sh stable-diffusion-stability-ai https://github.com/Stability-AI/stablediffusion.git cf1d67a6fd5ea1aa600c4df58e5b47da45f6bdbf \
  && rm -rf assets data/**/*.png data/**/*.jpg data/**/*.gif

RUN . /clone.sh CodeFormer https://github.com/sczhou/CodeFormer.git 8392d0334956108ab53d9439c4b9fc9c4af0d66d \
  && rm -rf assets inputs

RUN . /clone.sh BLIP https://github.com/salesforce/BLIP.git 3a29b7410476bf5f2ba0955827390eb6ea1f4f9d
RUN . /clone.sh k-diffusion https://github.com/crowsonkb/k-diffusion.git ab527a9a6d347f364e3d185ba6d714e22d80cb3c
RUN . /clone.sh clip-interrogator https://github.com/pharmapsychotic/clip-interrogator 2cf03aaf6e704197fd0dae7c7f96aa59cf1b11c9
RUN . /clone.sh generative-models https://github.com/Stability-AI/generative-models 477d8b9a7730d9b2e92b326a770c0420d00308c9
```

Seems like nothing broke except I had to update line 57 for clip-interrogator:
```Dockerfile
RUN mkdir ${ROOT}/interrogate && cp ${ROOT}/repositories/clip-interrogator/clip_interrogator/data/* ${ROOT}/interrogate
```

**Comment by cloudaxes** [comment]
@alex-red: I like what you did.  Though in my personal opinion, it's simpler to match the commits of dependencies that stable-diffusion-webui is using, rather than always updating to the latest.  The point is to make a container for that project, not do anything original.

Taming transformers was removed as a dependency of stable-diffusion-webui in this commit: https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/0b0f60f9549f067791418c085ac2a306a147c8d3.  So it can be removed as a dependency  here.

Automatic1111 did not update codeformers or BLIP, even though there's been work done on those repositories.

Automatic1111 did update Stability-AI/generative-models, but to a commit that is now not the latest.

I've incorporated your suggested changes into the latest commit in my pull request (#585 linked earlier) and updated only the repos: Stability-AI/stablediffusion, k-diffusion, clip-interrogator, and Stability-AI/generative-models, and removed taming-transformers.

**Comment by AbdBarho** [comment]
Thank you all for your efforts, and @cloudaxes for the PR, 1.6.0 is now merged!
https://github.com/AbdBarho/stable-diffusion-webui-docker/pull/585
